litmus-portal/backend/graphql-server/server.go

```diff
@@ -1,30 +1,44 @@
 package main
 
 import (
+	"github.com/99designs/gqlgen/graphql/handler"
+	"github.com/99designs/gqlgen/graphql/playground"
 	"github.com/gorilla/mux"
 	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/graph"
 	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/graph/generated"
-	store "github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/data-store"
-	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/database"
-	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/handlers"
+	database "github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/database/mongodb"
+	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/file_handlers"
 	"log"
 	"net/http"
-
-	"github.com/99designs/gqlgen/graphql/handler"
-	"github.com/99designs/gqlgen/graphql/playground"
+	"os"
 )
 
 const defaultPort = "8080"
 
+var err error
+
+func init() {
+	if os.Getenv("MONGODB_SERVER") == "" || os.Getenv("DB_NAME") == "" || os.Getenv("CLUSTER_JWT_SECRET") == "" || os.Getenv("SERVICE_ADDRESS") == "" {
+		log.Fatal("Some environment variable are not setup")
+	}
+
+	if err = database.DBInit(); err != nil {
+		log.Fatal(err)
+	}
+}
+
 func main() {
-	port := defaultPort
-	database.DBInit()
```

Comment:
Earlier it is not checking the err return from the DBInit Function

Reply:
Yes, I was throwing an error inside that function itself.

Reply:
We have to throw an error at a single place, as the other functions are returning the error.